modules/ai/assets/js/editor/pages/form-layout/index.js

```diff
@@ -141,6 +158,30 @@ const FormLayout = ( { onClose, onInsert, onData, onSelect, onGenerate, DialogHe
 		};
 	};
 
+	/**
+	 * @param {Attachment[]} items
+	 */
+	const onAttach = ( items ) => {
+		items.forEach( ( item ) => {
+			if ( ! attachmentsTypes[ item.type ] ) {
+				throw new Error( `Invalid attachment type: ${ item.type }` );
+			}
+
+			const typeConfig = attachmentsTypes[ item.type ];
+
+			if ( ! item.previewHTML && typeConfig.previewGenerator ) {
+				typeConfig.previewGenerator( item.content ).then( ( html ) => {
+					item.previewHTML = html;
+
+					setAttachments( () => [ item ] );
```

Comment:
should it be appended?

```suggestion
					setAttachments( ( prev ) => [ ...prev, item ] );
```

Reply:
It should replace